File: backend/src/main/java/com/bookflow/company/service/CompanyService.java
package com.bookflow.company.service;

import com.bookflow.company.dto.request.CreateCompanyRequest;
import com.bookflow.company.dto.request.UpdateCompanyRequest;
import com.bookflow.company.dto.response.CompanyResponse;

import java.util.List;

public interface CompanyService {

    CompanyResponse create(CreateCompanyRequest request);

    CompanyResponse findById(Long id);

    List<CompanyResponse> findAll();

    List<CompanyResponse> findAllCompanies();

    CompanyResponse update(Long id, UpdateCompanyRequest request);

    void delete(Long id);

    void activate(Long id);
}

File: frontend/src/services/companyService.js
import api from './api'

export const companyService = {
  create: (data) =>
    api.post('/api/v1/companies', data),

  getAll: () =>
    api.get('/api/v1/companies'),

  getById: (id) =>
    api.get(`/api/v1/companies/${id}`),

  update: (id, data) =>
    api.put(`/api/v1/companies/${id}`, data),

  delete: (id) =>
    api.delete(`/api/v1/companies/${id}`),

  activate: (id) =>
    api.patch(`/api/v1/companies/${id}/activate`),
}
